python/feathub/processors/spark/dataframe_builder/over_window_utils.py

```diff
@@ -0,0 +1,207 @@
+        for descriptor in agg_descriptors
+    }
+
+
+def _get_over_window_agg_select_expr(
+    expr: str,
+    over_window_descriptor: "OverWindowDescriptor",
+    agg_func: AggFunc,
+    window_spec: WindowSpec,
+) -> Column:
+
+    if (
+        over_window_descriptor.limit is not None
+        and over_window_descriptor.window_size is not None
+    ):
+        # TODO Adds aggregations on both limited and timed window
+        ...
+
+    if agg_func == AggFunc.AVG:
+        result = functions.avg(expr)
+    elif agg_func == AggFunc.MIN:
+        result = functions.min(expr)
+    elif agg_func == AggFunc.MAX:
+        result = functions.max(expr)
+    elif agg_func == AggFunc.SUM:
+        result = functions.sum(expr)
+    elif agg_func == AggFunc.FIRST_VALUE:
+        result = functions.first(expr)
+    elif agg_func == AggFunc.LAST_VALUE:
+        result = functions.last(expr)
+    elif agg_func == AggFunc.ROW_NUMBER:
+        result = functions.row_number()
+    elif agg_func == AggFunc.COUNT:
+        result = functions.count(expr)
+    elif agg_func == AggFunc.VALUE_COUNTS:
+        # TODO Adds VALUE_COUNTS support for SparkProcessor
+        raise FeathubTransformationException(
+            "VALUE_COUNTS is not supported for SparkProcessor currently."
+        )
+    else:
+        raise FeathubTransformationException(
+            f"Unsupported aggregation for SparkProcessor {agg_func}."
+        )
+
+    return result.over(window_spec)
+
+
+def evaluate_over_window_transform(
+    dataframe: DataFrame,
+    window_descriptor: "OverWindowDescriptor",
+    agg_descriptors: List["AggregationFieldDescriptor"],
+) -> DataFrame:
+
+    window_spec = _get_spark_over_window(window_descriptor)
+
+    if window_descriptor.filter_expr is not None:
+        agg_table = dataframe.filter(
+            functions.expr(window_descriptor.filter_expr)
+        ).withColumns(
+            _get_over_window_agg_column_list(
+                window_descriptor, agg_descriptors, window_spec
+            )
+        )
+
+        # For rows that do not satisfy the filter predicate, set the feature col
+        # to NULL.
+        null_cols = {}
+        for descriptor in agg_descriptors:
+            null_cols[descriptor.field_name] = functions.lit(None)
+        null_feature_table = dataframe.filter(
+            ~functions.expr(window_descriptor.filter_expr)
+        ).withColumns(null_cols)
+
+        # After union, order of the row with same grouping key is not preserved. We
+        # can only preserve the order of the row with the same grouping keys and
+        # filter condition.
+        result_table = agg_table.unionAll(null_feature_table)
+    else:
+        result_table = dataframe.withColumns(
+            _get_over_window_agg_column_list(
+                window_descriptor, agg_descriptors, window_spec
+            )
+        )
+    result_table.explain()
+    return result_table
+
+
+def _get_spark_over_window(
```

Comment:
It seems that this method does not do anything specific to the `over` logic. How about renaming it as `_get_spark_window_spec`?